casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -56,6 +60,32 @@ class RuntimeManager private (val initStateHash: ByteString, runtimeContainer: S
     result
   }
 
+  def computeBonds(hash: StateHash): Seq[Bond] = {
+    val resetRuntime = getResetRuntime(hash)
+    // TODO: Switch to a read only name
+    val bondsChannel     = Channel(Quote(Par().copy(exprs = Seq(Expr(GString("proofOfStake"))))))
+    val bondsChannelData = resetRuntime.space.getData(bondsChannel)
+    runtimeContainer.put(resetRuntime)
+    toBondSeq(bondsChannelData)
+  }
+
+  private def toBondSeq(data: Seq[Datum[ListChannelWithRandom]]): Seq[Bond] = {
+    assert(data.length == 1)
```

Comment:
Rather than asserts, these conditions should be included in the validation logic. We don't want the node to crash because there are multiple things on the proof of stake channel or the map is mis-formatted

Reply:
The failures will be caught by the Try in bondsCache so the node wouldn't crash.

Reply:
Ah, yes, you're right.